### ISSUE TITLE
Set interval dialog box should not close on selecting an item

### ISSUE THREAD

**Issue by sunkup** [ArnyminerZ/icsx5#84]
Right now the "Set interval" dialog box closes straight away when selecting a list item. The "Cancel" and "OK" buttons have effectively the same function - they only dismiss/close the dialog. 

I think we should either: 
- keep it as it is (dialog closes straight away on list item selection), but then show only one button, at the bottom, saying "close" or similar
- keep the buttons "Cancel" and "OK", but not close the dialog when the user changes his selection

My preference lies with the second option. What is your opinion? @ArnyminerZ 

Note: Not really important right now, but I want to track it for later fine tuning :)

**Comment by ArnyminerZ** [comment]
I agree with you. It's pretty stupid to have two buttons with the same functionality.

I'd also prefer to have two buttons:
- Cancel (does the same as if the user taps outside of the dialog): dismiss the dialog without storing the change
- Confirm the new interval: would save the preference and close the dialog.
